tutorials/0_What_is_ROS.ipynb

```diff
@@ -0,0 +1,150 @@
+    "- A node can publish to multiple topics\n",
+    "- A node can subscribe to multiple topics\n",
+    "\n",
+    "This allows you to be very flexible and adaptable with communication. But these nodes must be able to find each other in order to communicate with each other. For this purpose, a so called '*ROS master*' is used.\n",
+    "\n",
+    "## ROS master\n",
+    "The role of the ROS master is to enable individual ROS nodes to locate one another. It provides naming and registration services to the nodes in the ROS system. It tracks publishers and subscribers to topics as well as services (a tool for one-to-one communication in ROS). Once the nodes have located each other via the ROS master, they communicate with each other peer-to-peer. \n",
+    "The ROS master is started by using the **roscore** or **roslaunch** command. \n",
+    "\n",
+    "![](images/ros_master.png)\n",
+    "\n",
+    "The above image is a rough representation of the communication set up. First, the ROS master is activated. The nodes will, when activated, register to the master. The master knows which nodes are publishing on which topics and which nodes want to subscribe to which topic. The master in turn communicates back to the nodes so that a peer-to-peer connection can be established.  \n",
+    "\n",
+    "A more detailed explanation is represented by the figure below.\n",
+    "\n",
+    "![](images/ros_master_detailed.png)\n",
+    "\n",
+    "Once a ROS master is started an address is connected to it. The nodes need to know the network address of the master on startup to be able to connect to it. That is why the address is initially hard coded to be localhost:11311.\n",
+    "When a node is started, a random port number is assigned to it. For example the publisher node will be publishing a topic named /topic on localhost:4321. When the Subscriber node is started, it tells the ROS master that it wants to subscribe to the topic named /topic.\n",
+    "The master in turn tells the subscriber node that this topic is published at localhost:4321. Nextn the publisher and subscriber establish a connection in order to exchange data. \n",
+    "Take note that the master does not handle any information being sent. It only initializes the connection between publisher and subscriber. \n",
+    "\n",
+    "This publish/subscribe model is many-to-many one-way communication. This is not appropriate for **request/reply** interactions which are often required in a distributed system. Such a **request/reply** communication in ROS is achieved via a **service**. A service is defined by a pair of messages, one for request and one for reply. A node can register a specific service with the ROS master just as it registers/subscribes to its messages. \n",
+    "\n",
+    "For example, let's say we have a <u>Camera node</u> which collects data from a camera and we have an <u>Image processing node</u> to process the images. This situation is depicted in the figure below. \n",
```

Comment:
The HTML tag for underline is not parsed correctly.

Reply:
It seems to render fine on my machine and the VM ?

Reply:
I think it is not rendering on the GitHub viewer for some reason.